src/ExpansionPanel/ExpansionPanel.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { codes } from 'keycode';
+import { Card, Typography, IconButton } from '../';
+import Collapse from '../transitions/Collapse';
+import createStyleSheet from '../styles/createStyleSheet';
+import withStyles from '../styles/withStyles';
+
+export const styleSheet = createStyleSheet('MuiExpansionPanel', theme => ({
+  root: {
+    margin: 0,
+    transition: theme.transitions.create('margin', {
+      duration: theme.transitions.duration.shortest,
+    }),
+  },
+  disabled: {
+    backgroundColor: theme.palette.grey[200],
+    color: theme.palette.text.disabled,
+  },
+  expanded: {
+    margin: [theme.spacing.unit * 2, 0],
+    '&:first-child': {
+      marginTop: 0,
+    },
+    '&:last-child': {
+      marginBottom: 0,
+    },
+  },
+  header: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: 48,
+    transition: theme.transitions.create('height', {
+      duration: theme.transitions.duration.shortest,
+    }),
+    padding: [0, theme.spacing.unit * 3],
+  },
+  headerFocused: {
+    backgroundColor: theme.palette.grey[300],
+  },
+  headerExpanded: {
+    height: 64,
+  },
+  headerHover: {
+    '&:hover': {
+      cursor: 'pointer',
+    },
+  },
+  headerDisabled: {
+    opacity: 0.38,
+  },
+  expandButton: {
+    width: 40,
+    height: 40,
+    marginRight: -theme.spacing.unit * 2,
+    transform: 'rotate(0deg)',
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shortest,
+    }),
+  },
+  expandButtonOpen: {
+    transform: 'rotate(180deg)',
+  },
+  focusHolder: {
+    position: 'absolute',
+    '&:focus': {
+      outline: 'none',
+    },
+  },
+}));
+
+const enableTabbing = '0';
+const disableTabbing = '-1';
+
+class ExpansionPanel extends Component {
+  static defaultProps = {
+    disabled: false,
+    expandIcon: null,
+    onChange: () => {},
```

Comment:
I doubt the performance overhead is meaningful, but I agree with your first bullet.